tests/test_end_to_end.py

```diff
@@ -0,0 +1,104 @@
+"""
+These tests are run against a throw-away Notion account with a few pre-written
+pages. Since this is a throw-away account, we're fine including the auth_token
+in the codebase. The login for this throw-away account is in the Innolitics'
+1password "Everyone" vault. If new test pages are added, this will need to be
+used to create them.
+"""
+import sys
+import os.path
+from io import StringIO
+
+import yaml
+try:
+    from yaml import CLoader as Loader
+except ImportError:
+    from yaml import Loader
+
+from tests.utils import NOTION_ACCESS_TOKEN, parse_yaml_front_matter
+from n2y.main import main
+
+
+def run_n2y(arguments):
+    old_stdout = sys.stdout
+    sys.stdout = StringIO()
+    status = main(arguments, NOTION_ACCESS_TOKEN)
+    captured_stdout = sys.stdout.getvalue()
+    sys.stdout = old_stdout
+    return status, captured_stdout
+
+
+def test_simple_database_to_yaml():
+    '''
+    The database can be seen here:
+    https://fresh-pencil-9f3.notion.site/176fa24d4b7f4256877e60a1035b45a4
+    '''
+    object_id = '176fa24d4b7f4256877e60a1035b45a4'
+    status, stdoutput = run_n2y([object_id, '--output', 'yaml'])
+    assert status == 0
+    unsorted_database = yaml.load(stdoutput, Loader=Loader)
+    database = sorted(unsorted_database, key=lambda row: row["Name"])
+    assert len(database) == 3
+    assert database[0]["Name"] == "A"
+    assert database[0]["Tags"] == ["a", "b"]
+    assert database[0]["content"] is None
+
+
+# TODO: add test that dumps a database into a set of markdown files
+
+
+def test_all_blocks_page_to_markdown(tmp_path):
+    '''
+    The page can be seen here:
+    https://fresh-pencil-9f3.notion.site/Test-Page-5f18c7d7eda44986ae7d938a12817cc0
+    '''
+    object_id = '5f18c7d7eda44986ae7d938a12817cc0'
+    status, document_as_markdown = run_n2y([object_id, '--media-root', str(tmp_path)])
+    lines = document_as_markdown.split('\n')
+    metadata = parse_yaml_front_matter(document_as_markdown)
+    assert metadata['title'] == 'All Blocks Test Page'
+
+    # TODO: look into why there's extra space in between the list entries
+    assert status == 0
+    assert "Text block" in lines
+    assert "-   [ ] To do list block" in lines
+    assert "# Heading 1" in lines
+    assert "## Heading 2" in lines
+    assert "### Heading 3" in lines
+    assert "-   List block" in lines
+    assert "1.  Number list block" in lines
+    # TODO: add more blocks to the document, along with assertions
+
+    # "Unknown.jpeg" is a file block in the Notion page
+    assert os.path.exists(tmp_path / "Unknown.jpeg")
```

Comment:
I ordinarily agree we should avoid slow IO stuff in our tests, but since this is an end-to-end test and we're already making dozens of API calls, it seemed like avoiding saving the file isn't going to make much of a difference (and it will make the test a bit harder to read).

The file is cleaned up automatically by purest per https://docs.pytest.org/en/6.2.x/tmpdir.html#the-default-base-temporary-directory

Let me know if you're okay leaving this as is.